File: 1884-minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.java
class Solution {
    public int minOperations(String s) {
        int prev1 = 0, prev2 = 1;
        int ans1 = 0, ans2 = 0;
        for(int i=0; i<s.length(); i++) {
            int curr = s.charAt(i) - '0';
            if(prev1 != curr) {
                ans1++;
            }
            if(prev2 != curr) {
                ans2++;
            }
            prev1 = prev1 ^ 1;
            prev2 = prev2 ^ 1;
        }
        return Math.min(ans1, ans2);
    }
}

File: 1884-minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
    let prev1 = 0, prev2 = 1;
    let ans1 = 0, ans2 = 0;
    for(let ch of s) {
        let c = ch - '0';
        if(c !== prev1) {
            ans1++;
        }
        if(c !== prev2) {
            ans2++;
        }
        prev1 ^= 1;
        prev2 ^= 1;
    }
    return Math.min(ans1, ans2);
};